src/java_tools/buildjar/java/com/google/devtools/build/java/turbine/turbine_benchmark.sh

```diff
@@ -64,9 +64,12 @@ function reset_and_modify_file() {
 export -f reset_and_modify_file
```

Comment:
With that change I'm seeing larger numbers, but I'm also not seeing the expected difference from enabling PGO.

I suspect I'm just holding something wrong, I'll keep digging.

```
===== Benchmarking prebuilt Turbine =====
...
Benchmark 1: build_target
  Time (mean ± σ):     21.216 s ±  0.281 s    [User: 0.006 s, System: 0.019 s]
  Range (min … max):   20.928 s … 21.574 s    5 runs
...
===== Benchmarking Turbine built from HEAD =====
...
Benchmark 1: build_target
  Time (mean ± σ):     21.090 s ±  0.245 s    [User: 0.013 s, System: 0.013 s]
  Range (min … max):   20.705 s … 21.298 s    5 runs
```

Reply:
Is that on an arm64 Mac? This is similar to what I was seeing on mine (almost no improvement), but the numbers on x86_64 Linux are much better (at least on my machine).

Reply:
That was on my x86_64 Linux machine, it's interesting we're seeing different results there.

Reply:
I see a speedup on an m1:

```
===== Benchmarking prebuilt Turbine =====
Benchmark 1: build_target
  Time (mean ± σ):     28.852 s ±  0.680 s    [User: 0.012 s, System: 0.016 s]
  Range (min … max):   28.024 s … 29.773 s    5 runs
===== Benchmarking Turbine built from HEAD =====
Benchmark 1: build_target
  Time (mean ± σ):     26.594 s ±  1.154 s    [User: 0.012 s, System: 0.016 s]
  Range (min … max):   25.105 s … 28.237 s    5 runs
```

Reply:
I also get surprising results on my machine now in some benchmark executions, which didn't happen when I originally tested. Since I had to introduce a workaround for an issue with the disk cache, I will double-check that the benchmark actually works and then try it out on more machines.

Switching this to draft in the meantime.

Reply:
Thanks, to be clear I'm also fine with merging this and iterating on the configuration (or the benchmark), but I'll hold off for now until I hear how the additional benchmarks go.

Reply:
I did some more experiments and summarized them in the PR description. In total, I found that this improves CPU time spent on Turbine very significantly (almost 2x on an M3 Max), but on machines with many CPU cores this doesn't translate into an improvement of wall time on the benchmark.

This makes me positive that merging this is an improvement, so I'll mark this as ready for review.

Do you know of a good way to accurately capture CPU time of the Bazel build performed by the benchmark?

Reply:
> I did some more experiments and summarized them in the PR description. In total, I found that this improves CPU time spent on Turbine very significantly (almost 2x on an M3 Max), but on machines with many CPU cores this doesn't translate into an improvement of wall time on the benchmark.
> 
> This makes me positive that merging this is an improvement, so I'll mark this as ready for review.

Thanks! I'll get the merge started.

> Do you know of a good way to accurately capture the CPU time of the Bazel build performed by the benchmark?

I do not have a suggestion for this off-hand.

Reply:
I updated the commit message again: When directly benchmarking the Turbine binary, switching to Oracle GraalVM makes the biggest difference compared to also adding PGO. When considering wall time on my machine before, PGO was the biggest win (probably due to more fortunate scheduling of the expensive javac actions).